src/firefox/index.js

```diff
@@ -251,7 +251,37 @@ export async function useProfile(
     customPrefs = {},
   }: UseProfileParams = {},
 ): Promise<FirefoxProfile> {
-  const profile = new FirefoxProfile({destinationDirectory: profilePath});
+  let profile;
+  const finder = new FirefoxProfile.Finder();
+  const finderGetPath = promisify(finder.getPath.bind(finder));
```

Comment:
`finderGetPath` will reject with an Error when the Finder instance is not able to find the requested profile name, which is something that we expect to happen (because the user could have no profile with that name yet).

To be able to only call `findGetPath` when we know that it will resolve to the path instead of being rejected with an Error, we can use the finder.readProfiles method and check if the named profile that we need actually exists before trying to resolve them into paths, e.g. something like the following snippet should do the trick:

```js
  ...
  const finder = new FirefoxProfile.Finder(searchProfilesPath);
  const finderGetPath = promisify(finder.getPath, finder);
  const finderReadProfiles = promisify(finder.readProfiles, finder);

  // Read the profiles list from the profiles.ini file from `searchProfilesPath`.
  await finderReadProfiles();

  // Helper function which returns true if the profile name exists in the profiles.ini file.
  const hasProfileName = (profileName) => {
    return finder.profiles.filter(
      (profileDef) => profileDef.Name === profileName
    ).length !== 0;
  };

  ...
  const defaultProfilePath = (
    hasProfileName('default') && await finderGetPath('default')
  );
  const defaultDevProfilePath = (
    hasProfileName('default') && await finderGetPath('dev-edition-default')
  );
```

This way `defaultProfilePath` and `defaultDevProfilePath` will be the profile path (which is a string), or `undefined` if the profile doesn't exist in the `profiles.ini` list.